### ISSUE TITLE
Docker Desktop 4.19: Fresh installation fails: System.Runtime.InteropServices.COMException

### ISSUE THREAD

**Issue by HarleyDavidson86** [docker/for-win#13509]
- [x] I have tried with the latest version of Docker Desktop
  - [ ] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Actual behavior

Installation of Docker desktop fails

### Expected behavior

Installation works fine

### Information

  - Windows Version: Windows 10 (20H2) Build 19042.1052
  - Docker Desktop Version: 4.19
  - WSL2 or Hyper-V backend? WSL2
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: No

I checked the following steps:
- There is no wsl machine running
- I downloaded the exe several times
- I disabled antivirus programs
- I cleaned the whole registry from keys with "docker" in it
- I deleted all directories with "docker" in it under APPDATA folder

```
C:\Users\dsust\Downloads>start /w "" "Docker Desktop Installer.exe" install --backend=wsl-2
-------------------------------------------------------------------------------->8
Version: 4.19.0 (106363)
Sha1:
Started on: 2023.05.25 12:33:49.089
Resources: C:\Users\dsust\Downloads\resources
OS: Windows 10 Pro
Edition: Professional
Id: 2009
Build: 19042
BuildLabName: 19041.1.amd64fre.vb_release.191206-1406
File: C:\ProgramData\DockerDesktop\install-log-admin.txt
CommandLine: "Docker Desktop Installer.exe"  install --backend=wsl-2
You can send feedback, including this log file, at https://github.com/docker/for-win/issues
[2023-05-25T12:33:51.412936700Z][ManifestAndExistingInstallationLoader][I] No install path specified, looking for default installation registry key
[2023-05-25T12:33:51.422931900Z][Installer][I] No installation found
[2023-05-25T12:33:51.476935700Z][InstallWorkflow][I] Cancel pending background download
[2023-05-25T12:33:51.502936600Z][BackgroundTransfer][I] Cancel current background transfer job
[2023-05-25T12:33:55.717573600Z][InstallWorkflow][I] Using package: res:DockerDesktop
[2023-05-25T12:33:55.718564900Z][InstallWorkflow][I] Downloading
[2023-05-25T12:33:56.861080400Z][InstallWorkflow][I] Extracting manifest
[2023-05-25T12:33:57.497738800Z][InstallWorkflow][I] Manifest found: version=106363, displayVersion=4.19.0, channelUrl=https://desktop.docker.com/win/main/amd64/appcast.xml
[2023-05-25T12:33:57.497738800Z][InstallWorkflow][I] Checking prerequisites
[2023-05-25T12:33:57.616844500Z][InstallWorkflow][I] Prompting for optional features
[2023-05-25T12:33:57.832856100Z][Installer][W] Failed to track the installer started event
[2023-05-25T12:34:11.454168700Z][InstallWorkflow][I] Selected backend mode: wsl-2
[2023-05-25T12:34:11.455168900Z][InstallWorkflow][I] Unpacking artifacts
[2023-05-25T12:36:07.956581000Z][InstallWorkflow][I] Deploying component CommunityInstaller.CreateGroupAction
[2023-05-25T12:36:14.807928700Z][InstallWorkflow][I] Deploying component CommunityInstaller.AddToGroupAction
[2023-05-25T12:36:21.669268300Z][InstallWorkflow][I] Deploying component CommunityInstaller.EnableFeaturesAction
[2023-05-25T12:36:21.725991500Z][InstallWorkflow-EnableFeaturesAction][I] Required features: VirtualMachinePlatform, Microsoft-Windows-Subsystem-Linux
[2023-05-25T12:36:21.829478700Z][InstallWorkflow][E] Installation failed System.Exception: Component CommunityInstaller.EnableFeaturesAction failed:  ---> System.Runtime.InteropServices.COMException
   bei System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   bei System.Management.ManagementObjectCollection.ManagementObjectEnumerator.MoveNext()
   bei System.Linq.Enumerable.<CastIterator>d__97`1.MoveNext()
   bei System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   bei System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   bei System.Linq.Enumerable.Any[TSource](IEnumerable`1 source)
   bei CommunityInstaller.EnableFeaturesAction.GetFeaturesToEnable()
   bei CommunityInstaller.EnableFeaturesAction.<DoAsync>d__45.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   bei CommunityInstaller.InstallWorkflow.<DoHandleD4WPackageAsync>d__30.MoveNext()
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei CommunityInstaller.InstallWorkflow.<DoHandleD4WPackageAsync>d__30.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei CommunityInstaller.InstallWorkflow.<DoProcessAsync>d__23.MoveNext()
[2023-05-25T12:36:21.832047900Z][InstallWorkflow][I] Rollbacking component CommunityInstaller.AddToGroupAction
[2023-05-25T12:36:21.833009800Z][InstallWorkflow][I] Rollbacking component CommunityInstaller.CreateGroupAction
[2023-05-25T12:36:21.833009800Z][InstallWorkflow][I] Rollbacking component CommunityInstaller.UnpackArtifactsStep
[2023-05-25T12:36:21.841051300Z][FileSystem][I] Deleting C:\Program Files\Docker\Docker
[2023-05-25T12:36:22.215266600Z][FileSystem][I] Deleted C:\Program Files\Docker\Docker successfully
[2023-05-25T12:36:22.215266600Z][InstallWorkflow][I] Rollbacking component CommunityInstaller.ExtractManifestStep
[2023-05-25T12:36:22.218271500Z][InstallWorkflow][I] Rollbacking component CommunityInstaller.DownloadStep
```

### Output of `& "C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" check`

Folder is empty.

### Steps to reproduce the behavior
Double click on the exe file or call it via `start /w "" "Docker Desktop Installer.exe" install --backend=wsl-2`

**Comment by bsousaa** [comment]
From the error log:  `Required features: VirtualMachinePlatform, Microsoft-Windows-Subsystem-Linux`

Check System requirements on Windows: https://docs.docker.com/desktop/install/windows-install/ Enable the WSL 2 feature on Windows.

**Comment by HarleyDavidson86** [comment]
Hi @bsousaa 
the command `wsl --install` shows me the helptext, which is fine according to MS documentation:

> The above command only works if WSL is not installed at all, if you run wsl --install and see the WSL help text, please try running wsl --list --online to see a list of available distros and run wsl --install -d <DistroName> to install a distro. To uninstall WSL, see [Uninstall legacy version of WSL](https://learn.microsoft.com/en-us/windows/wsl/troubleshooting#uninstall-legacy-version-of-wsl) or [unregister or uninstall a Linux distribution](https://learn.microsoft.com/en-us/windows/wsl/basic-commands#unregister-or-uninstall-a-linux-distribution).

I am getting many other strange errors with code 0x8000ffff (E_UNEXPECTED)... I think this is my first goal to solve before continuing with the installation of docker desktop.

**Comment by bsousaa** [comment]
It sounds like you have issues on your Windows setup then

**Comment by HarleyDavidson86** [comment]
I went to hard times to get the latest 22H2 windows updates receiving a 0x8000ffff error code.
I finally managed it to update my system successfully with the [Windows 10 Update Assistent Tool](https://support.microsoft.com/en-us/topic/windows-10-update-assistant-3550dfb2-a015-7765-12ea-fba2ac36fb3f).

And now I am on 22H2 and the installation of Docker Desktop 4.20 was successful.

Maybe this is helpful for other users facing this error code.
